packages/x-data-grid/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -208,19 +208,22 @@ function extractColumnWidths(
       const header = findGridHeader(apiRef.current, column.field);
       if (header) {
         const title = header.querySelector(`.${gridClasses.columnHeaderTitle}`);
+        const titleContainer = header.querySelector(`.${gridClasses.columnHeaderTitleContainer}`)!;
         const content = header.querySelector(`.${gridClasses.columnHeaderTitleContainerContent}`)!;
         const iconContainer = header.querySelector(`.${gridClasses.iconButtonContainer}`);
         const menuContainer = header.querySelector(`.${gridClasses.menuIcon}`);
         const element = title ?? content;
-
         const style = window.getComputedStyle(header, null);
+        const titleContainerStyle = window.getComputedStyle(titleContainer, null);
         const paddingWidth = parseInt(style.paddingLeft, 10) + parseInt(style.paddingRight, 10);
+        const flexGap = iconContainer && !menuContainer ? parseInt(titleContainerStyle.gap, 10) : 0;
```

Comment:
2px gap is only considered when column is sortable and column menu is disabled in other cases it is 0.

Also, I want to simplify this way of calculating the width of column header by targeting the wrappers instead of exact elements but it didn't seem to work out well. 
@romgrk any opinion here?

Reply:
Thoughts https://github.com/mui/mui-x/issues/15854#issuecomment-2537923992 ?